js/app.js

```diff
@@ -0,0 +1,139 @@
+'use-strict';
+//constructor function for images
+var total_clicks = 26;
+var all_mall_items = [];
+
+var currently_displayed_right_product;
+var currently_displayed_center_product;
+var currently_displayed_left_product;
+
+var product_container = document.getElementById('product_container');
+
+var left_img = document.getElementById('left_img');
+var center_img = document.getElementById('center_img');
+var right_img = document.getElementById('right_img');
+
+var left_h2 = document.getElementById('left_h2');
+var center_h2 = document.getElementById('center_h2');
+var right_h2 = document.getElementById('right_h2');
+
+
+var Mall_item = function(item_name, filepath) {
+  this.item_name = item_name;
+  this.filepath = filepath;
+  this.num_times_shown = 0;
+  this.num_times_clicked = 0;
+  all_mall_items.push(this);
+};
+
+var render_products = function(product, target_img, target_h2){
+  target_img.src = product.filepath;
+  target_h2.textContent = product.item_name;
+};
+
+var handle_click_on_item = function(event) {
+  console.log('click');
+  if(event.target.tagName === 'IMG'){
+    if(event.target.id === 'left_img'){
+      currently_displayed_left_product.num_times_clicked ++;
+    } else if(event.target.id === 'center_img')
+    {currently_displayed_center_product.num_times_clicked ++;
+    } else if (event.target.id === 'right_img'){
+      currently_displayed_right_product.num_times_clicked ++;
+    }//need to tell it to not register clicks in other part of section
+
+    total_clicks --;
+
+    //currently pulls from all_mall_items array to display randomly.
+    //can I splice this array?
+
+    var left_img_idx = Math.floor(Math.random() * all_mall_items.length);
+    var center_img_idx = Math.floor(Math.random() * all_mall_items.length);
+    var right_img_idx = Math.floor(Math.random() * all_mall_items.length);
+
+    /*trying to splice the array being fed into all_mall_items, to start preventing duplicates
+   I'm keeping this here but commented out. Whenever I try to splice the array, it breaks the program because

```

Comment:
This is a perfect way to approach it. If you use splice, it removes it from the original though and you are correct, eventually you will have "deleted" all your products with splice. One way to circumvent that would be to create another global var : temp_all_mall_items and give it the same products as all_mall_items, when there are not enough products left in temp_all_mall_items, you can refill it with all the products again